lib/plaid/telemetry.ex

```diff
@@ -0,0 +1,47 @@
+defmodule Plaid.Telemetry do
+  @moduledoc """
+  Custom implementation of `telemetry` using Tesla Middleware.
+  """
+  @events_prefix [:plaid, :request]
+
+  @behaviour Tesla.Middleware
+
+  @impl Tesla.Middleware
+  def call(env, next, _opts) do
+    metadata = env.opts[:metadata] || %{}
+    start_time = System.monotonic_time()
+
+    :telemetry.execute(@events_prefix ++ [:start], %{system_time: start_time}, metadata)
+
+    try do
+      Tesla.run(env, next)
+    rescue
+      exception ->
+        :telemetry.execute(
+          @events_prefix ++ [:exception],
+          %{duration: System.monotonic_time() - start_time},
+          Map.put(metadata, :exception, exception)
+        )
+
+        reraise exception, __STACKTRACE__
+    else
+      {:ok, env} = result ->
+        :telemetry.execute(
+          @events_prefix ++ [:stop],
+          %{duration: System.monotonic_time() - start_time},
```

Comment:
Are you sure this module is needed though? This is what https://github.com/teamon/tesla/blob/008b287da51620852c7564f369a90eecf92f5542/lib/tesla/middleware/telemetry.ex#L2 does, no?

Reply:
You're correct, it's almost exactly the same. The reason I chose to do a custom implementation is for backward compatibility with the existing code. Specifically, @kanes115  added events named [:plaid, :request, :start], whereas the Tesla out of the box middleware uses [:tesla, :request, :start], and unfortunately that cannot be overridden.

Reply:
Ahhh I see, I see, it is a breaking change, so this is your opportunity for it, maybe write documentation on how to use this version, but use the default tesla one.

So mark everything under the repo as `deprecated` and give some room for people to do so.

Reply:
I fixed this in the latest commit